docs/design/reserve-action.md

```diff
@@ -0,0 +1,152 @@
+it and not allowing the other tasks to be scheduled on the node.
+
+## Function Detail 
+
+### Job starving
+
+If the `pending` time of job is greater that the set time threshold, the task has the risk of starvation, and needs to
+reserve node resources for it. If there is unscheduled task under the job, but the job has already met the `running` 
+condition and is already `running`, we think that this job does not need to reserve node resource any more.
+
+### Node resource reservation rule
+
+If the allocatable resources on the node is greater that the resource request of the task, then the node satisfies the 
+scheduled requirement of the task now or later. This node can reserve resources for this task. If the idle resource of 
+the current node has met the scheduling condition of the task, resources will be allocated to this task, and the task 
+will be directly assigned to the node. If the idle resource of the node can not meet the resource requirement of the task,
+the node will reserve resources for this task. Once a node reserve resources for a task, no other tasks will be scheduled 
+on this node, for the node's allocatable resources will no longer meet the scheduling requirements of any task.
+
+A node may reserve resources only for one task, or may reserve resources for multiple tasks, depending on the size of the 
+allocatable resources of node.
+
+Because of that some of node resources was reserved for tasks and will not be used by other tasks, the resources 
+reservation will affect the task scheduling process of the entire cluster. In theory, the more nodes that participate in
+resource reservation, the better it is to solve the problem of the task being starved to death. However, the fewer nodes 
+participating in regular scheduling, and the greater the impact on regular cluster scheduling. Therefore, the number of 
+nodes participating in resource reservation is configurable.
+
+How to choose which nodes will be selected as candidate nodes for starving job scheduling. All nodes calculated a score
+according to the usage resources on the node, and then all nodes are sorted in descending order of rank, nodes with higher
+scores will be selected as candidate nodes.
+
+In the process of calculating node scores based on node resource usage, there will be cpu memory and scalar resources 
+participating in the score calculation. The formula for calculating score for a single resource is  
+
+<img src="http://chart.googleapis.com/chart?cht=tx&chl=\Large score=100-\frac{used}{allocatable}" style="border:none;">
+
+The formula for calculating the node score based on the above three resources is
+
+<img src="http://chart.googleapis.com/chart?cht=tx&chl=\Large score=\frac{\frac{cpuScore\pm memoryScore}{2}\pm2*sclarScore}{3}" style="border:none;">
+
+## Data Model
+
+Introduce one node resource `Reserved` resource, used to indicate the size of resources reserved on the node. The size 
```

Comment:
>Introduce one node resource `Reserved` resource,

one field?

Reply:
including `cpu` `memory` and `scalar` three resources fields